### PR TITLE
Adding the ability to specify custom next and prev arrow elements

### PR DESCRIPTION
**Pull request by ryanbagwell**
This allows you to pass in custom next and prev elements as props.

### CODE REVIEW

**Comment by akiran** [comment]
I will review this pull request soon

**Comment by ryanbagwell** [comment]
Thanks. I new to react, so I'm not sure if this is the best way to accomplish this. It requires something like this:

````
var prevArrow = React.createClass({

  render: function() {
    return <button {...this.props}>Previous</button>
  }

})

<Carousel prevArrow=prevArrow>
  <div>One</div>
  <div>Two</div>
  <div>Three</div>
</Carousel>
````

Note that ````this.props```` has to be passed to the element. At the least, I probably should have added an update to the documentation.

**Comment by ryanbagwell** [comment]
@akiran just wanted to see what you thought about this PR.

**Comment by akiran** [comment]
@ryanbagwell, Sorry. I couldn't get to this because of other tasks.
I will definitely review this PR and merge it this weekend.

**Comment by ryanbagwell** [comment]
awesome, thank you.

File: lib/default-props.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ var defaultProps = {
     variableWidth: false,
     vertical: false,
     // waitForAnimate: true,
-    afterChange: null
+    afterChange: null,
+    nextButton: null,
```
Comment (akiran) [inline lib/default-props.js]: @ryanbagwell  Can you change these names to "nextArrow" and "prevArrow"  to match the terminology of  slick carousel

File: lib/inner-slider.jsx [hunk 2]
```diff
@@ -127,8 +128,36 @@ var Slider = React.createClass({
...
+        key: '0',
+        ref: 'previous',
+        'data-role': "none",
+        className: classnames(prevClasses),
+        style: {display: 'block'},
+        onClick: prevHandler
+      }
+
+      if (this.props.prevArrow) {
+        var prevArrow = <this.props.prevArrow {...prevArrowProps} />;
```
Comment (akiran) [inline lib/inner-slider.jsx]: Here you used the prevArrow and nextArrow terminology correctly.
I think you made a mistake in default-props.js file.
Can you correct it.

**Comment by ryanbagwell** [comment]
@akiran sorry about that. I pushed a fix. Thanks for letting me know.

**Comment by akiran** [comment]
@ryanbagwell  Thanks for the pull request.

**Comment by chemoish** [comment]
Is there an example implementation of this?

**Comment by ryanbagwell** [comment]
I should have included one in the readme. You basically pass in react components as props with names prevArrow and nextArrow.

**Comment by chemoish** [comment]
I don't believe there is any documentation and I have yet to reverse engineer the source https://github.com/akiran/react-slick/commit/cfb1c3212c50e34a7e4df1b226447e86db9b488c.

I have tried

```js
// inline
<Slick nextArrow={<button></button>}></Slick>

// settings
var settings = {
  nextArrow={<button></button>}
};

<Slick {...settings}></Slick>
```

Seems that https://github.com/akiran/react-slick/blob/44ae518cd08fc4fd7d5523738591832a5e4d5711/src/arrows.jsx#L31 only accepts a string to modify the tag name (which doesn't seem that useful).

---

Anyways, I am looking into https://github.com/akiran/react-slick/blob/44ae518cd08fc4fd7d5523738591832a5e4d5711/src/inner-slider.jsx#L78, but figured I would ask.

**Comment by ryanbagwell** [comment]
Your example should work. Not sure if you need brackets around your button component.

**Comment by ryanbagwell** [comment]
@chemoish see my comment on March 24 and note the part about passing this.props to your button component.

**Comment by chemoish** [comment]
@ryanbagwell, I tried your code, but it doesn't seem to satisfy the use case (Correct me if I am wrong—Thanks for the response btw).

```js
let NextArrow = React.createClass({
    render() {
        return (
            <button className="slick-arrow" {...this.props}>Next</button> // need a slick-arrow class
        );
    }
});
```

The current implementation allows you to override the `tag` as well as the `innerHTML`. It also allows you to add default props (Both inline or configured).

However, it doesn't allow you to extend properties that are hard coded in https://github.com/akiran/react-slick/blob/44ae518cd08fc4fd7d5523738591832a5e4d5711/src/arrows.jsx#L69.

---

Solution 1: Copy pasta https://github.com/akiran/react-slick/blob/44ae518cd08fc4fd7d5523738591832a5e4d5711/src/arrows.jsx#L42 and pass that in as a prop (Bad).

Solution 2: ???

**Pull request merged**